app/services/platform-apps/index.ts

```diff
@@ -385,22 +386,19 @@ export class PlatformAppsService extends
   }
 
   unloadApps() {
-    this.state.loadedApps.forEach(app => {
-      this.REMOVE_APP(app.id);
-      this.appUnload.next(app.id);
-    });
-    localStorage.removeItem(this.localStorageKey);
-    // TODO: navigate to live if on that topnav tab
-
-    if (this.devServer) {
-      this.devServer.stopListening();
-      this.devServer = null;
-    }
+    this.state.loadedApps.forEach(app => this.unloadApp(app));
   }
 
-  unloadApp(appId: string) {
-    this.REMOVE_APP(appId);
-    this.appUnload.next(appId);
+  unloadApp(app: ILoadedApp) {
+    this.REMOVE_APP(app.id);
+    this.appUnload.next(app.id);
```

Comment:
It's better to trigger this event at the end of this method. When the app has been completely removed

Reply:
+1